### ISSUE TITLE
Improve error codes returned to user

### ISSUE THREAD

**Issue by ayaleloehr** [department-of-veterans-affairs/prescriptions-api#46]
It seems like almost all errors from the Rx API return a 400 (documented here: https://github.com/department-of-veterans-affairs/prescriptions-api/blob/master/api-doc.md)

I don't think all these are really 400's, which is for a bad request. We should return better errors

**Comment by saneshark** [comment]
100% agree, I think this is going to take at least a little bit of thought as we probably want to do our best to come up with a normalized set of error codes that we can leverage across various integrated apis. 

I'll elaborate for the sake of posterity. For example, just because MHV provides us with a status code 400, minor code 101, and some description of the code -- it does not mean we *should* return the same result. Instead, it would be better for us to return the appropriate status for not authenticated - usually 401 or 403. Similarly following https://en.wikipedia.org/wiki/List_of_HTTP_status_codes for other error codes.

**Comment by ayaleloehr** [comment]
@saneshark is now working on a common gem that I believe is focused initially on error codes. reminder to add that back in here when stable.

**Comment by saneshark** [comment]
Yes, making some progress on that front, hope to have something release worthy in next 24 hours. That being said, based on conversation today -- curious to see if we are also going to be moving everything into the vets-api, making the need for "common gem" less necessary.
